src/lib/containers/Facets.js

```diff
@@ -0,0 +1,216 @@
+                    if (element.facetType === "enumeration") {
+                        let children = []
+                        element.facetValues.forEach(
+                            facetValue => {
+                                let key = uuidv4()
+                                let checked = false
+                                let chosen = this.state.selectedFacets[element.columnName]
+                                if (chosen && chosen.facetValues.indexOf(facetValue.value) !== -1) {
+                                    checked = true
+                                }
+                                children.push(
+                                    <div key={key}>
+                                        <input checked={checked} onChange={this.handleClick({name: element.columnName, value: facetValue.value, isSelected: checked})} id={key} type="checkbox"/>
+                                        <label htmlFor={key}>{facetValue.value + ` (${facetValue.count})`}</label>
+                                    </div>
+                                )
+                            }
+                        )
+                        let name = <strong> {element.columnName} </strong>
+                        let formatted = (<div key={uuidv4()}>
+                                            {name}
+                                            {children.map(child => {return child})}
+                                        </div>)
+                        structuredRender.push(formatted)
+                    }
+                }
+            )
+            return (<div>
+                        {
+                            structuredRender.map(
+                                element => {
+                                    return element
+                                }
+                            )
+                        }
+                    </div>)
+        }
+    }
+
+    // https://medium.freecodecamp.org/reactjs-pass-parameters-to-event-handlers-ca1f5c422b9
+    handleClick = (dict) => (event) => {
+        let selectedFacets = cloneDeep(this.state.selectedFacets)  // deep copy
+
+        // if there is no entry for this column name into the selection of facets
+        if (!selectedFacets.hasOwnProperty(dict.name)) {
+            let newEntry = {
+                columnName: dict.name,
+                concreteType: "org.sagebionetworks.repo.model.table.FacetColumnValuesRequest",
+                facetValues: []
+            }
+            selectedFacets[dict.name] = newEntry
+        }
+
+        // grab the facet values assoicated for this column
+        let specificFacet = selectedFacets[dict.name]
+        // if its not selected then we add as having been chosen, otherwise we 
+        // have to delete it
+        if (!dict.isSelected) {
+            specificFacet.facetValues.push(dict.value)
+        } else {
+            specificFacet.facetValues = specificFacet.facetValues.filter(
```

Comment:
not a better way to remove an item from an array?  you found "splice", which looks more straightforward (use indexOf to find the element)